src/Sulu/Component/Content/Repository/ContentRepository.php

```diff
@@ -890,4 +922,16 @@ private function resolveHasChildren(Row $row)
 
         return \count(\iterator_to_array($result->getRows())) > 0;
     }
+
+    public function supportsDescendantType(string $type): bool
+    {
+        try {
+            $class = new \ReflectionClass($type);
+        } catch (\ReflectionException $e) {
+            // in case the class does not exist there is no support
+            return false;
+        }
+
+        return $class->implementsInterface(SecurityBehavior::class);
```

Comment:
We should avoid reflection and use here the [is_a](https://www.php.net/manual/de/function.is-a.php) function of php:

```php
return is_a($type, SecurityBehavior::class, true)
```

Reply:
Think `is_a` is better in performance then doing a reflection and think we should avoid reflection if not really needed. Maybe we should then also update the other code snippets or does is_a handle something different?

Reply:
From what I know the fear of reflections being slow comes from a pre-PHP7 era, and in some PHP version they got much faster. Therefore I would prefer to do a benchmark before changing all occurences based on a guess. We can change this occurence including all others in a separate PR, if the difference is really that big.